lib/galaxy/tool_util/deps/dependencies.py

```diff
@@ -80,7 +82,11 @@ class DependenciesDescription:
     other potential remote execution mechanisms.
     """
 
-    def __init__(self, requirements=[], installed_tool_dependencies=[]):
+    def __init__(self, requirements=None, installed_tool_dependencies=None):
+        if requirements is None:
```

Comment:
@nsoranzo do you know why this was here needed and not elsewhere?

Reply:
@bgruening `requirements = requirements or []` would replace a `ToolRequirements` instance of length 0 with `[]`, which then fails when calling the `resolvable()` method on it. After investigating, I propose a better fix in https://github.com/galaxyproject/galaxy/pull/11044 .